pkgs/development/python-modules/generic/default.nix

```diff
@@ -166,7 +116,8 @@ python.stdenv.mkDerivation (builtins.removeAttrs attrs ["disabled"] // {
        tmp_path=$(mktemp -d)
```

Comment:
While we're at it, could we add logic that only performs these steps if the setup.py is for the package we're trying to install? The current method creates very strange behavior if you run `nix-shell` for one package in another package's folder. Something along the lines of:

```bash
# Check if setup.py exists
if test -e setup.py; then
  # Check that the name of the package we're building matches the one in setup.py
  if echo ${name} | grep -q $(${python.interpreter} setup.py --name); then
    ...
  fi
fi
```